core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/util/DatabaseUtil.java

```diff
@@ -571,71 +572,100 @@ public static Connection getDBConnection(DataSource dataSource) throws SQLExcept
     }
 
     public static void closeConnection(Connection dbConnection) {
-
-        if (dbConnection != null) {
-            try {
-                dbConnection.close();
-            } catch (SQLException e) {
-                log.error("Database error. Could not close statement. Continuing with others. - " + e.getMessage(), e);
-            }
-        }
+        close(dbConnection);
     }
 
-    private static void closeResultSet(ResultSet rs) {
-
-        if (rs != null) {
+    /**
+     *
+     * @param dbObject -  java.sql object types that implement both Wrapper and AutoCloseable interfaces
+     */
+    private static <AutoClosableWrapper extends AutoCloseable & Wrapper> void close(AutoClosableWrapper dbObject) {
+        if (dbObject != null) {
             try {
-                rs.close();
+                dbObject.close();
+            } catch (SQLRecoverableException ex) {
```

Comment:
Does jtds , and microsoft drivers throw this exception (I suppose not). if thats the case you might have to resort to sqlstate class code.